### PR TITLE
[CSE-99] Support tx fees. Bump up cardano-sl.

### PR DESCRIPTION
**Pull request by pva701**
* Support tx fees
* Bump up to last `cardano-sl`:
** Remove `NotificationMode`, use `ReaderT` instead (because of we can't have several `MonadReader` in our monad stack now...)
** Simplify `Transformer.hs`
** Rewrite `Main.hs` and `ExplorerOptions.hs` with according to `cardano-sl/Main.hs`

FYI: I didn't test it, because it's required to compile purescript

### CODE REVIEW

File: src/Pos/Explorer/Web/Server.hs [hunk 1]
```diff
@@ -156,6 +158,11 @@ explorerHandlers _sendActions =
 ----------------------------------------------------------------
 -- API Functions
 ----------------------------------------------------------------
+rights' :: [(Either a b, c)] -> [(b, c)]
```
Comment (ksaric) [inline src/Pos/Explorer/Web/Server.hs]: I don't really understand what this is for? Doesn't `rights` already take care of this behaviour:
```
rights   :: [Either a b] -> [b]
rights x = [a | Right a <- x]
```

**Comment by ksaric** [comment]
@pva701 Please wait until I merge `cardano-sl-0.5` in `master` so you can try this again. I'll do it today hopefully.

**Comment by pva701** [comment]
@ksaric ok

**Comment by ksaric** [comment]
@pva701 Done. If you want me to take a look at the conflicts, I can.

**Comment by pva701** [comment]
@ksaric np, I think I will resolve it by myself

**Comment by pva701** [comment]
@ksaric merged

File: src/Pos/Explorer/Web/Server.hs [hunk 2]
```diff
@@ -223,19 +215,18 @@ getBlocksPage mPageNumber pageSize = do
...
-        -> m (Block SscGodTossing)
-    getBlockOrThrow headerHash = DB.blkGetBlock headerHash >>=
-        maybeThrow (Internal "Block with hash cannot be found!")
-
+-- Either get the block from the @HeaderHash@ or throw an exception.
+getBlundOrThrow
+    :: (DB.MonadBlockDB SscGodTossing m, MonadThrow m)
+    => HeaderHash
+    -> m (Blund SscGodTossing)
+getBlundOrThrow headerHash = DB.blkGetBlund headerHash >>=
```
Comment (ksaric) [inline src/Pos/Explorer/Web/Server.hs]: Why a `Blund`? We only save `Block` `HeaderHash`, we don't need `Blund` here?

Comment (pva701) [inline src/Pos/Explorer/Web/Server.hs]: we need it because of `toBlockEntry` requires it

File: src/Pos/Explorer/Socket/App.hs [hunk 3]
```diff
@@ -123,60 +123,60 @@ notifierServer settings connVar = do
             route [("/socket.io", handler)]
 
 periodicPollChanges
-    :: forall ssc m.
-       (ExplorerMode m, SscHelpersClass ssc)
+    :: forall ssc ctx m.
+       (ExplorerMode ctx m, SscHelpersClass ssc)
```
Comment (gromakovsky) [inline src/Pos/Explorer/Socket/App.hs]: I am looking at it and just want to say that `ssc` parameter doesn't seem to make sense here. You are using `SscGodTossing` anyway,

It's not related to this PR, just saying.

**Pull request merged**